### DISCUSSION TITLE
Relevancy behavior with synonyms

### DISCUSSION THREAD

**Discussion by curquiza** [meilisearch/product#436] [Feedback & Feature Proposal]
Following: https://github.com/meilisearch/meilisearch/discussions/2309

Currently, when adding synonyms to Meilisearch, documents containing the synonyms will have the same "weight" than documents containing the original word

Example:
If my synonym is `{ "NYC": "new york city" }`, if I type `NYC` in the search bar, documents containing `NYC` will not necessarily be put before the documents containing `new york city`. There is not prioritization in terms of relevancy between the original word and the synonym. 

[This user](https://github.com/meilisearch/meilisearch/discussions/2309) seems to want to prioritize the original word compared to the synonyms regarding the relevancy.
 
We need more use cases and feedback to consider this feature request since we don't know if the current behavior is expected by most of the users.

Also I ping @ManyTheFish for the technical details. Is it even doable?

**Comment by ManyTheFish** [comment]
Hello @curquiza,
If we speak about `multi-word synonyms`, `word-split`, and `ngrams`, it is already possible to force Meilisearch to rank them by putting the [Exactness](https://docs.meilisearch.com/learn/core_concepts/relevancy.html#_6-exactness) ranking rule upper in the list. 
I think the optimal position would the between `Typo` and `Proximity`:
```diff
[
      "words",
      "typo",
+    "exactness",
      "proximity",
      "attribute",
      "sort",
-     "exactness",
]
```

However, this will not work for 1-word synonyms like `{ "football": "soccer" }`, the Exactness ranking doesn't make any difference between both words.
Technically, stopping counting 1-word synonyms as exact matches in Exactness is straightforward and could be a `good first issue`.
